sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobDownloadAsyncResponse.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.models;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.rest.ResponseBase;
+import com.azure.storage.blob.HttpGetterInfo;
+import com.azure.storage.common.implementation.StorageImplUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+/**
+ * This class contains the response information returned from the server when downloading a blob.
+ *
+ * <p>
+ * Failed downloads are automatically retied based on the {@link ReliableDownloadOptions download options}, the retry
+ * will be resumed from the point where the download failed. This allows for the download to be consumed as one
+ * continuous stream.
+ * </p>
+ */
+public final class BlobDownloadAsyncResponse extends ResponseBase<BlobDownloadHeaders, Flux<ByteBuffer>> {
+    private final ReliableDownloadOptions options;
+    private final HttpGetterInfo info;
+    private final Function<HttpGetterInfo, Mono<BlobDownloadAsyncResponse>> getter;
+
+    /**
+     * Constructs a {@link BlobDownloadAsyncResponse}.
+     *
+     * @param request Request sent to the service.
+     * @param statusCode Response status code returned by the service.
+     * @param headers Raw headers returned in the response.
+     * @param value Stream of download data being returned by the service.
+     * @param deserializedHeaders Headers deserialized into an object.
+     * @param options {@link ReliableDownloadOptions} used to retry downloads is errors occur.
+     * @param info {@link HttpGetterInfo} used to retry a partially failed download from a specific offset.
+     * @param getter Function used to attempt to retry a partially failed download.
+     */
+    public BlobDownloadAsyncResponse(HttpRequest request, int statusCode, HttpHeaders headers, Flux<ByteBuffer> value,
+        BlobDownloadHeaders deserializedHeaders, ReliableDownloadOptions options, HttpGetterInfo info,
+        Function<HttpGetterInfo, Mono<BlobDownloadAsyncResponse>> getter) {
```

Comment:
Can the getter method be made internal to this? This was previously a package private constructor which mean it should only be called by internal code.